rasa/core/policies/rule_policy.py

```diff
@@ -987,3 +1174,70 @@ def _metadata(self) -> Dict[Text, Any]:
     @classmethod
     def _metadata_filename(cls) -> Text:
         return "rule_policy.json"
+
+    def get_rule_only_slots_loops(self) -> Tuple[List[Text], List[Text]]:
+        """Gets the lists of slots and loops that are used only in rule data.
+
+        Returns:
+            A tuple of lists of slots and loops that are used only in rule data.
+        """
+        return (
+            self.lookup.get(RULE_ONLY_SLOTS, []),
+            self.lookup.get(RULE_ONLY_LOOPS, []),
+        )
+
+
+class RulePolicyPrediction(PolicyPrediction):
```

Comment:
why do we need this?

Reply:
I didn't want to add prediction source to general `PolicyPrediction` object, since it is only used inside `RulePolicy`

Reply:
ok